activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -358,33 +358,33 @@ def insert_fixture(fixture, table_name)
       def insert_fixtures(fixtures, table_name)
         return if fixtures.empty?
 
-        columns = schema_cache.columns_hash(table_name)
+        execute(build_fixture_sql(fixtures, table_name), "Fixtures Insert")
+      end
+
+      def insert_fixtures_set(fixture_set, tables_to_delete = [])
+        total_sql = []
 
-        values = fixtures.map do |fixture|
-          fixture = fixture.stringify_keys
+        fixture_set.each do |table_name, fixtures|
```

Comment:
How about something more like:

```ruby
fixture_inserts = fixture_set.map { |table_name, fixtures| build_fixture_sql(fixtures, table_name) }
total_sql = combine_multi_statements(fixture_inserts)
```

Where `combine_multi_statements` in abstract is `[statements.join]`, and all the details of *why* some statements may not be combinable (i.e., there are length limits) can be pushed down to the MySQL adapter.